tensordict/tensordict.py

```diff
@@ -5400,3 +5467,22 @@ def _clone_value(value, recurse):
         return value.clone(recurse=False)
     else:
         return value
+
+
+def detect_loop(tensordict: TensorDict) -> bool:
+    visited = set()
+    visited.add(id(tensordict))
+
+    def detect(t_d: TensorDict):
+        for k, v in t_d.items():
+            if id(v) in visited:
+                return True
+            visited.add(id(v))
```

Comment:
Can move the `visited.add` and `visited.remove` inside the if statement, as it's only relevant in the tensordict case.